### PR TITLE
Feature: smtp data source plugin

### LINKED ISSUE #2262: SMTP server data source plugin
### Feature
Add a data source to send email from the Tooljet app.
- User should be able to add their SMTP server

### Operations
- Send an Email with Plain text

  Properties:
   - Text content
   
- Send an Email with HTML

  Properties:
   - HTML content
#### Common properties
- From
- To
- Email subject
  
#### Fields
 - Host
 - Port
 - User
 - Password

 ### If the feature is approved, would you be willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by shah21**
Resolves #2262

### CODE REVIEW

File: plugins/packages/smtp_server/lib/manifest.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    "name": "SMTP Server",
+    "kind": "smtp_server",
+    "exposedVariables": {
+      "isLoading": false,
+      "data": {},
+      "rawData": {}
+    },
+    "options": {
+      "host": {
+        "type": "string"
```
Comment (gsmithun4) [inline plugins/packages/smtp_server/lib/manifest.json:17]: Type is no longer used, you can remove them and keep the keys which needs to be encrypted.
```
      "user": {
        "encrypted":true
      },
      "password": {
        "encrypted": true
      }
```

File: plugins/packages/smtp/lib/index.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+export default class Smtp implements QueryService {
+  async run(sourceOptions: SourceOptions, queryOptions: QueryOptions, dataSourceId: string): Promise<QueryResult> {
+    const nodemailerTransport = await this.getConnection(sourceOptions);
+
+    const from = queryOptions.from;
+    const from_name = queryOptions.from_name;
+    const to = queryOptions.to;
+    const subject = queryOptions.subject;
+    const textContent = queryOptions.textContent;
+    const htmlContent = queryOptions.htmlContent;
```
Comment (gsmithun4) [inline plugins/packages/smtp/lib/index.ts:14]: try
`const { from, to, from_name, subject, textContent, htmlContent } = queryOptions`

File: plugins/packages/smtp/lib/index.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      },
+      to,
+      subject,
+      text: textContent,
+      html: htmlContent,
+      attachments: attachments && filesData(attachments),
+    };
+ 
+    try {
+      await new Promise((resolve,reject)=>nodemailerTransport.sendMail(mailOptions, (err:any,success:any)=>{
```
Comment (gsmithun4) [inline plugins/packages/smtp/lib/index.ts:37]: Will it work?
`const info = await nodemailerTransport.sendMail`

File: plugins/packages/smtp/lib/index.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  async testConnection(sourceOptions: SourceOptions): Promise<ConnectionTestResult> {
+    const transporter = await this.getConnection(sourceOptions);
+  
+    if (!transporter) {
+      throw new Error('Invalid credentials');
+    }
+
+    try{
+      await new Promise((resolve,reject)=>{
+        transporter.verify(function (error:any, success:any) {
```
Comment (gsmithun4) [inline plugins/packages/smtp/lib/index.ts:63]: try
`await transporter.verify()`

File: plugins/packages/smtp/lib/index.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    return {
+      status: 'ok',
+    };
+  }
+
+  async getConnection(sourceOptions: SourceOptions, _options?: object): Promise<any> {
+    const host = sourceOptions.host;
+    const port = Number(sourceOptions.port);
+    const user = sourceOptions.user;
+    const pass = sourceOptions.password;
```
Comment (gsmithun4) [inline plugins/packages/smtp/lib/index.ts:86]: Same here
https://github.com/ToolJet/ToolJet/pull/2263/files#diff-1c3ffc487b03352fa9690f06499a9d46acbc7b50446e47efdaec897a7cf33a0bR14

File: plugins/packages/smtp/lib/manifest.json [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    "name": "SMTP",
+    "kind": "smtp",
+    "exposedVariables": {
+      "isLoading": false,
+      "data": {},
+      "rawData": {}
+    },
+    "options": {
+      "user": {
+        "encrypted":true
```
Comment (gsmithun4) [inline plugins/packages/smtp/lib/manifest.json:17]: Do we need to encrypt username?

File: docs/docs/data-sources/smtp.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+
+# smtp
+
+ToolJet can connect to SMTP Server databases to read and write data. 
+
+- [smtp](#smtp)
+  - [Connection](#connection)
+  - [Querying SMTP Server](#querying-smtp-server)
+
+## Connection
```
Comment (akshaysasidrn) [inline docs/docs/data-sources/smtp.md:10]: Please update the docs

File: docs/docs/data-sources/smtp.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
+
+# smtp
+
+ToolJet can connect to SMTP Server databases to read and write data.
```
Comment (akshaysasidrn) [inline docs/docs/data-sources/smtp.md:4]: ToolJet can connect to SMTP server to send emails.

Comment (shah21) [reply docs/docs/data-sources/smtp.md]: It's auto-generated content. I have added a new doc just a minute ago

File: docs/docs/data-sources/smtp.md (hunk 7 above)
Comment (shah21) [reply docs/docs/data-sources/smtp.md:6]: Done

**Review by akshaysasidrn — APPROVED** [review]

**Review by gsmithun4 — APPROVED** [review]
LGTM

**Pull request merged**